File: gdx2d-desktop/src/hevs/gdx2d/lib/utils/GdxGraphicsTest.java
package hevs.gdx2d.lib.utils;

import com.badlogic.gdx.graphics.Color;

import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

/**
 * Torture test for graphics primitives 
 * @author Marc Pignat (pim)
 * 
 * Test all grapics primitives in different order, colors, zoom...
 */
public class GdxGraphicsTest extends PortableApplication {

	public GdxGraphicsTest(boolean onAndroid) {
		super(onAndroid, 512, 512);
	}

	public static void main(String args[]) {
		new GdxGraphicsTest(false);
	}

	float zoom = 1.0f;
	boolean zoom_up = true;
	
	@Override
	public void onInit() {
		
	}
	
	void manage_zoom()
	{
		if (zoom_up)
		{
			zoom *= 1.01;
			if (zoom > 1.2)
			{
				zoom_up = false;
			}
		}
		else
		{
			zoom *= 0.99;
			if (zoom < 0.8)
			{
				zoom_up = true;
			}
		}
	}

	@Override
	public void onGraphicRender(GdxGraphics g) {
		
		manage_zoom();
		
		g.clear();
		g.zoom(zoom);
		
		for (int i = 0 ; i < 20 ; i++)
		{
			g.setPixel(40+i, 0+i, Color.BLUE);
		}
		g.drawCircle(10, 10, 10);
		g.drawFilledCircle(30, 10, 10, Color.RED);
		
		for (int i = 0 ; i < 20 ; i++)
		{
			g.setPixel(60-i, 0+i, Color.BLUE);
		}
		
		g.drawRectangle(70, 10, 10, 10, 30);
		g.drawFilledRectangle(90, 10, 10, 10, 60, Color.CYAN);
	}

}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/perf_tests/GdxGraphicsTest.scala
package ch.hevs.gdx2d.demos.perf_tests

import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.graphics.Color

/**
 * Torture test for graphics primitives. Tests all graphic primitives in
 * different orders, colors and zoom levels.
 *
 * @author Marc Pignat (pim)
 */
class GdxGraphicsTest extends DesktopApplication {

  private var zoomLevel: Float = 1.0f
  private var zoomUp: Boolean = true

  override def onInit(): Unit = {
    setTitle(getClass.getSimpleName)
  }

  private def manageZoom(): Unit = {
    if (zoomUp) {
      zoomLevel *= 1.01f
      if (zoomLevel > 1.2f) zoomUp = false
    } else {
      zoomLevel *= 0.99f
      if (zoomLevel < 0.8f) zoomUp = true
    }
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    manageZoom()
    g.clear()
    g.zoom(zoomLevel)

    for (i <- 0 to 19) g.setPixel((40 + i).toFloat, i.toFloat, Color.BLUE)
    g.drawCircle(10f, 10f, 10f)
    g.drawFilledCircle(30f, 10f, 10f, Color.RED)

    for (i <- 0 to 19) g.setPixel((60 - i).toFloat, i.toFloat, Color.BLUE)

    g.drawRectangle(70f, 10f, 10f, 10f, 30f)
    g.drawFilledRectangle(90f, 10f, 10f, 10f, 60f, Color.CYAN)
    g.drawSchoolLogoUpperRight()
    g.drawFPS()
  }
}

object GdxGraphicsTest {
  def main(args: Array[String]): Unit = new GdxGraphicsTest().launch()
}
